pkg/controllers/disruption/helpers.go

```diff
@@ -191,6 +191,45 @@ func GetCandidates(ctx context.Context, cluster *state.Cluster, kubeClient clien
 	return lo.Filter(candidates, func(c *Candidate, _ int) bool { return shouldDeprovision(ctx, c) }), nil
 }
 
+// buildDisruptionBudgets will return a map for nodePoolName -> numAllowedDisruptions and an error
+func buildDisruptionBudgets(ctx context.Context, cluster *state.Cluster, clk clock.Clock, kubeClient client.Client) (map[string]int, error) {
+	nodePoolList, err := nodepoolutil.List(ctx, kubeClient)
+	if err != nil {
+		return nil, fmt.Errorf("listing node pools, %w", err)
+	}
+	numNodes := map[string]int{}
+	deleting := map[string]int{}
+	disruptionBudgetMapping := map[string]int{}
+	// We need to get all the nodes in the cluster
+	// Get each current active number of nodes per nodePool
+	// Get the max disruptions for each nodePool
+	// Get the number of deleting nodes for each of those nodePools
+	// Find the difference to know how much left we can disrupt
+	nodes := cluster.Nodes()
+	for _, node := range nodes {
+		if !node.Managed() {
+			continue
+		}
+		nodePool := node.Labels()[v1beta1.NodePoolLabelKey]
+		if node.MarkedForDeletion() {
+			deleting[nodePool]++
+		}
+		numNodes[nodePool]++
+	}
+
+	for i := range nodePoolList.Items {
+		nodePool := nodePoolList.Items[i]
+		disruptions, err := nodePool.GetAllowedDisruptions(ctx, clk, numNodes[nodePool.Name])
+		if err != nil {
+			// Don't return here as we shouldn't block all disruption within the cluster on one misconfigured nodepool.
+			logging.FromContext(ctx).With("nodePool", nodePool.Name).Debugf("getting allowed disruptions, %w", err)
```

Comment:
```suggestion
			logging.FromContext(ctx).With("nodepool", nodePool.Name).Errorf("getting allowed disruptions, %v", err)
```

Shouldn't this be an error? Since we should scream about this not being correct?

Reply:
Good catch!